crates/bevy_window/src/lib.rs

```diff
@@ -119,6 +120,31 @@ impl Plugin for WindowPlugin {
         if settings.close_when_requested {
             app.add_system(close_when_requested);
         }
+
+        // Register types
+        app.register_type::<WindowResized>()
+            .register_type::<CreateWindow>()
+            .register_type::<RequestRedraw>()
+            .register_type::<WindowCreated>()
+            .register_type::<WindowCloseRequested>()
+            .register_type::<WindowClosed>()
+            .register_type::<CursorMoved>()
+            .register_type::<CursorEntered>()
+            .register_type::<CursorLeft>()
+            .register_type::<ReceivedCharacter>()
+            .register_type::<WindowFocused>()
+            .register_type::<WindowScaleFactorChanged>()
+            .register_type::<WindowBackendScaleFactorChanged>()
+            .register_type::<FileDragAndDrop>()
+            .register_type::<WindowMoved>();
+        app.register_type::<WindowId>()
+            .register_type::<PresentMode>()
+            .register_type::<WindowResizeConstraints>()
+            .register_type::<WindowMode>()
+            .register_type::<WindowPosition>()
+            .register_type::<MonitorSelection>()
+            .register_type::<WindowDescriptor>();
+        app.register_type::<PathBuf>();
```

Comment:
This should probably go somewhere more centralized like `bevy_core`, no?

Reply:
I called that out as an open question. A lot of types that could be registered in `CorePlugin` aren't being registered there, so I wasn't sure whether to put this over there or leave it here where it's being used. I'm fine either way, just thought it would be good to get input from others.

Reply:
This should be fine. I think eventually I'd like to make `App::register_type` to not overwrite by default, which is the main concern I have.